network/transport/v2/handlers.go

```diff
@@ -236,12 +236,10 @@ func (p *protocol) handleTransactionList(peer transport.Peer, envelope *Envelope
 	data := handlerData{}
 
 	// TODO convert to trace logging
-	log.Logger().Infof("handling handleTransactionList from peer (peer=%s, conversationID=%s)", peer.ID.String(), cid.String())
+	log.Logger().Infof("handling handleTransactionList from peer (peer=%s, conversationID=%s, message=%d/%d)", peer.ID.String(), cid.String(), msg.MessageNumber, msg.TotalMessages)
 
 	// check if response matches earlier request
-	var conversation *conversation
-	var err error
-	if conversation, err = p.cMan.check(envelope, data); err != nil {
+	if _, err := p.cMan.check(envelope, data); err != nil {
```

Comment:
we could add an additional check in the checkResponse method for this message; `currentMsg.number == prevMsg.number + 1` (store message & total number in conversation)

Reply:
GRPC should deliver the messages in the correct order, so if the numbering is off this would indicate a mistake in another implementation. If this is by mistake there is no real harm, and if this is on purpose (repeatedly using the same low number) these checks don't really solve anything since there is no limit on `TotalMessages`

Reply:
and it would trigger missing prevs and thus log an error indicating the other node is doing something wrong.